lib/src/Pos/Node/API.hs

```diff
@@ -547,47 +551,130 @@ instance ToSchema (V1 Version) where
         pure $ NamedSchema (Just "V1Version") $ mempty
             & type_ .~ SwaggerString
 
+
+newtype SecurityParameter = SecurityParameter Int
+    deriving (Show, Eq, Generic, ToJSON, FromJSON, Arbitrary)
+
+instance Buildable SecurityParameter where
+    build (SecurityParameter i) = bprint shown i
+
+instance ToSchema SecurityParameter where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "SecurityParameter") $ mempty
+            & type_ .~ SwaggerString
+
+
+instance ToSchema (V1 Core.SlotId) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Core.SlotId") $ mempty
+            & type_ .~ SwaggerString
+
+deriving instance ToJSON (V1 Core.SlotId)
+deriving instance FromJSON (V1 Core.SlotId)
+
+instance Arbitrary (V1 Core.SlotId) where
+    arbitrary = fmap V1 arbitrary
+
+
+instance ToSchema Byte where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Byte") $ mempty
+            & type_ .~ SwaggerString
+
+instance Buildable Byte where
+    build b = bprint shown b
+
+
+instance Arbitrary (V1 Core.TxFeePolicy) where
+    arbitrary = fmap V1 arbitrary
+
+instance ToSchema (V1 Core.TxFeePolicy) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Core.TxFeePolicy") $ mempty
+            & type_ .~ SwaggerString
+
+deriving instance ToJSON (V1 Core.TxFeePolicy)
+deriving instance FromJSON (V1 Core.TxFeePolicy)
+
+
+instance Arbitrary (V1 Core.SlotCount) where
+    arbitrary = fmap V1 arbitrary
+
+instance ToSchema (V1 Core.SlotCount) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Core.SlotCount") $ mempty
+            & type_ .~ SwaggerString
+
+deriving instance ToJSON (V1 Core.SlotCount) -- TODO ?
+deriving instance FromJSON (V1 Core.SlotCount) --TODO ?
+
 -- | The @static@ settings for this wallet node. In particular, we could group
 -- here protocol-related settings like the slot duration, the transaction max size,
 -- the current software version running on the node, etc.
 data NodeSettings = NodeSettings
-    { setSlotDuration   :: !SlotDuration
-    , setSoftwareInfo   :: !(V1 Core.SoftwareVersion)
-    , setProjectVersion :: !(V1 Version)
-    , setGitRevision    :: !Text
+    { setSlotId            :: !(V1 Core.SlotId)
+    , setSlotDuration      :: !SlotDuration
+    , setSlotCount         :: !(V1 Core.SlotCount)
+    , setSoftwareInfo      :: !(V1 Core.SoftwareVersion)
+    , setProjectVersion    :: !(V1 Version)
+    , setGitRevision       :: !Text
+    , setMaxTxSize         :: !Byte
```

Comment:
A bit rough at the API-level. A value with units would be a bit more frirendly (cf `SlotDuration` for instance)